### PR TITLE
feat(chip-set): add delete all chips button

### PR DESCRIPTION
**Pull request by maltja**
fix Lundalogik/crm-feature#1329

## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/chip-set/chip-set.tsx [hunk 1]
```diff
@@ -295,6 +304,8 @@ export class ChipSet {
                     'force-invalid': this.isInvalid(),
                 }}
                 onClick={this.handleTextFieldFocus}
+                onMouseEnter={this.onMouseEnter}
+                onMouseLeave={this.onMouseLeave}
```
Comment (jgroth) [inline src/components/chip-set/chip-set.tsx:308]: I don't think we need these :slightly_smiling_face: 
We should be able to hide the button with only CSS rules and not have to rely on any JS code

**Review by jgroth — CHANGES_REQUESTED** [review]

**Review by jgroth — DISMISSED** [review]
Good work! :confetti_ball:

**Review by jgroth — APPROVED** [review]

**Comment by adrianschmidt** [comment]
@jgroth Methinks you didn't test keyboard navigation for this… 😉

You can tab to the clear-button, but there doesn't seem to be any way to actually activate it. I've tried both \<enter\>  and \<space\>.

image

Also, if you don't hover the component when tabbing to the clear-button, it looks a bit broken:

image

I'd be fine with just taking this button out of the tab-flow, since you can already delete the chips by pressing backspace when the input is focused, and unless you have an unreal amount of chips selected, that's pretty fast and easy. That also avoids the downside of having to tab twice to get past this component. What do you say @Kiarokh?

Regardless, it shouldn't be possible to tab to the button when there are no chips selected.

**Comment by adrianschmidt** [comment]
Found some more small issues:

When disabled, it's still possible to tab to the button:

image

When read-only, it's not only possible to tab to the button, but it also becomes visible when any chip is hovered:

image

(There's also a small issue with the label cutting part of the chips, both when disabled and when read-only, but I don't think that's got anything to do with this PR. @Kiarokh: If I recall correctly, you moved the label a little bit a while back? It might have happened then. Can you take a look at it?)

**Comment by Kiarokh** [comment]
@adrianschmidt I think:
* user should be able to tab to the (x) button
* (x) should be none-existing when there are no chips as you said
* (x) should have a `title` and `area-label`

**Comment by adrianschmidt** [comment]
Ok. I'll add the above in an issue.

**Comment by jgroth** [comment]
> @jgroth Methinks you didn't test keyboard navigation for this… 

That is a fair assumption! :smile:

**Comment by Kiarokh** [comment]
@adrianschmidt I think we have addressed issues above here: https://github.com/Lundalogik/lime-elements/pull/938

**Comment by Kiarokh** [comment]
oops, except the keyboard interactions (**Enter** / Space)

**Pull request merged**